uarray/_backend.py

```diff
@@ -66,7 +85,13 @@ def unpickle_function(mod_name, qname, self_):
         raise UnpicklingError from e
 
 
-def pickle_function(func):
+@no_type_check
+def pickle_function(
+    func: Callable[..., Any],
+) -> tuple[
+    Callable[[str, str, object], Callable[..., Any]],
+    tuple[str, str, Any | None],
+]:
```

Comment:
The `try`/`except` blocks here are simply too complicated for type checkers to understand, unfortunately.